### PR TITLE
feat: wms v2

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Connector;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore;
+    using Be.Vlaanderen.Basisregisters.Utilities.HexByteConvertor;
+    using NetTopologySuite.Geometries;
+    using NetTopologySuite.IO;
+    using NodaTime;
+    using StreetName;
+    using StreetName.Events;
+
+    [ConnectedProjectionName("WMS adressen")] // TODO moeten we dit aanpassen naar V2 ?
```
Comment (jvandaal) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs:15]: @ArneD what should be the ConnectedProjectionName and ~Description?

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+                    {
+                        address.PositionMethod = ConvertGeometryMethodToString(message.Message.GeometryMethod);
+                        address.PositionSpecification = ConvertGeometrySpecificationToString(message.Message.GeometrySpecification);
+                        address.SetPosition(ParsePosition(message.Message.ExtendedWkbGeometry));
+                        UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                    },
+                    ct,
+                    updateHouseNumberLabelsBeforeAddressUpdate: true,
+                    updateHouseNumberLabelsAfterAddressUpdate: true,
+                    allowUpdateRemovedAddress: true);
```
Comment (jvandaal) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs:371]: We added allowUpdateRemovedAddress: true based upon how this was used in the V1 projections

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+                    message.Message.AddressPersistentLocalId,
+                    address =>
+                    {
+                        address.Removed = true;
+                        address.SetHouseNumberLabel(null);
+                        UpdateVersionTimestamp(address, message.Message.Provenance.Timestamp);
+                    },
+                    ct,
+                    updateHouseNumberLabelsBeforeAddressUpdate: true,
+                    allowUpdateRemovedAddress: true);
```
Comment (jvandaal) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs:403]: We added allowUpdateRemovedAddress: true based upon how this was used in the V1 projections

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
```
Comment (jvandaal) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs:418]: We added allowUpdateRemovedAddress: true based upon how this was used in the V1 projections

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs (hunk 1 above)
Comment (ArneD) [reply src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2Projections.cs:15]: same as V1

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    using System;
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.Utilities;
+    using Infrastructure;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.EntityFrameworkCore.Metadata.Builders;
+    using NetTopologySuite.Geometries;
+    using NodaTime;
+
+    public class AddressDetailItemV2
```
Comment (ArneD) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2.cs:12]: Please use `AddressWmsItem` and all dependent classes

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            PositionSpecification = positionSpecification;
+            Removed = removed;
+            VersionTimestamp = versionTimestamp;
+        }
+
+        public int AddressPersistentLocalId { get; set; }
+        public int StreetNamePersistentLocalId { get; set; }
+        public string? PostalCode { get; set; }
+        public string HouseNumber { get; set; }
+        public string? HouseNumberLabel { get; private set; }
```
Comment (ArneD) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2.cs:52]: Is there a scenario where this is `null`?

File: src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2.cs [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        public int? HouseNumberLabelLength { get; private set; }
```
Comment (ArneD) [inline src/AddressRegistry.Projections.Wms/AddressDetailV2/AddressDetailV2.cs:53]: Is there a scenario where this is null?

File: src/AddressRegistry.Projections.Wms/Migrations/20221010123905_Add_AddressDetailV2.cs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    public partial class Add_AddressDetailV2 : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "AddressDetailsV2",
+                schema: "wms.address",
+                columns: table => new
+                {
+                    AddressPersistentLocalId = table.Column<int>(type: "int", nullable: false)
```
Comment (ArneD) [inline src/AddressRegistry.Projections.Wms/Migrations/20221010123905_Add_AddressDetailV2.cs:18]: No identity

**Review by ArneD — CHANGES_REQUESTED** [review]
see comments

**Pull request merged**